interface/repository/csv_pokemon_repository.go

```diff
@@ -0,0 +1,148 @@
+package repository
+
+import (
+	"io"
+	"strconv"
+
+	"github.com/FernandoGal25/academy-go-q42021/domain/model"
+	customErrors "github.com/FernandoGal25/academy-go-q42021/error"
+	"github.com/FernandoGal25/academy-go-q42021/infrastructure/datastore"
+)
+
+/*
+	Repository that handles the operations of pokemon in a CSV file.
+*/
+type CSVPokemonRepository struct {
+	Handler *datastore.CSVHandler
+}
+
+/*
+	Returns an instance of CSVPokemonRepository.
+*/
+func NewCSVPokemonRepository(h *datastore.CSVHandler) CSVPokemonRepository {
+	return CSVPokemonRepository{h}
+}
+
+func buildPokemon(p *model.Pokemon, data []string, ID uint64) error {
+	p.ID = ID
+
+	p.Name = data[1]
+
+	h, err := strconv.Atoi(data[2])
+	if err != nil {
+		return customErrors.CSVFormatError{Message: "CSV ITEM DOES NOT POSSESS A VALID HEIGHT", Err: err}
```

Comment:
Could initialize a variable with this message, and use `fmt.Sprintf` to avoid repeating yourself.

``` go
const CSVItemMessage = "CVS item does not posses a valid %s attribute"
...
return customErrors.CSVFormatError{Message: fmt.Sprintf(CSVItemMessage, "heigth"), Err: err}
return customErrors.CSVFormatError{Message: fmt.Sprintf(CSVItemMessage, "weight"), Err: err}
return customErrors.CSVFormatError{Message: fmt.Sprintf(CSVItemMessage, "order"), Err: err}
return customErrors.CSVFormatError{Message: fmt.Sprintf(CSVItemMessage, "baseExperience"), Err: err}

// Or even better, generate a function in customErrors accepting both parameters, and return an instance of this error.

func GenerateCSVFormatErr(attribute string, err error) CSVFormatError {
	return customErrors.CSVFormatError{Message: fmt.Sprintf(CSVItemMessage, atribute), Err: err}
}

return customErrors.GenerateCSVFormatErr("height", err)
return customErrors.GenerateCSVFormatErr("weight", err)
return customErrors.GenerateCSVFormatErr("order", err)
return customErrors.GenerateCSVFormatErr("baseExperience", err)
```